### PR TITLE
[beta] Prepare Rust 1.70.0

### PR DESCRIPTION
**Pull request by emilyalbini**
This PR replaces the version placeholders, and changes the channel to `beta`.

r? @ghost
cc @rust-lang/release

### CODE REVIEW

**Comment by ehuss** [comment]
The error is likely caused by #109044.

File: tests/ui/feature-gates/test-listing-format-json.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// no-prefer-dynamic
+// compile-flags: --test
+// run-flags: --list --format json -Zunstable-options
+// run-fail
+// check-run-results
+// ignore-nightly
```
Comment (Mark-Simulacrum) [inline tests/ui/feature-gates/test-listing-format-json.rs:6]: Are we checking this on the stable builder today? Or does that not run tests - I forget.

**Review by Mark-Simulacrum — APPROVED** [review]
r=me

File: tests/ui/feature-gates/test-listing-format-json.rs (hunk 1 above)
Comment (pietroalbini) [reply tests/ui/feature-gates/test-listing-format-json.rs:6]: It should be running the full tests, so yes it should be executed in theory.

**Comment by ehuss** [comment]
BTW, I marked #110361 as beta-nominated to avoid the error that happened above.

**Pull request merged**